python/meeting-scheduling/src/meeting_scheduling/domain.py

```diff
@@ -0,0 +1,183 @@
+from dataclasses import dataclass, field
+from typing import List, Optional, Annotated, Union
+from timefold.solver.domain import (
+    planning_entity, planning_solution, PlanningId, PlanningVariable,
+    PlanningEntityCollectionProperty, ProblemFactCollectionProperty, ValueRangeProvider,
+    PlanningScore, PlanningPin
+)
+from timefold.solver import SolverStatus
+from timefold.solver.score import HardMediumSoftScore
+from .json_serialization import JsonDomainBase
+from pydantic import Field
+
+# Time granularity is 15 minutes (which is often recommended when dealing with humans for practical purposes).
+GRAIN_LENGTH_IN_MINUTES = 15
+
+@dataclass
```

Comment:
Note: this should be frozen so a `__hash__` will be added (alternatively, `unsafe_hash` can be set).
```suggestion
@dataclass(frozen=True)
```

Reply:
Noted. I will implement this with pydantic, directly.

Going back to the Java version to double-check, I noticed that, besides Person, the following classes implement equals()/hashcode():
- TimeGrain
- Room
- MeetingAssignment

Which would mean, from my understanding of pydantic docs:

```
class FooBarModel(BaseModel):
    model_config = ConfigDict(frozen=True)

    a: str
    b: dict
    
```
for all of them. Could you please confirm?

Reply:
You cannot make MeetingAssignment frozen, since that class is mutated at runtime (it is a planning entity). The rest can be frozen. Note this is the base class we use for Pydantic models: https://github.com/TimefoldAI/timefold-quickstarts/blob/cd69bab7fcc47e5334c71a32fb540ba3d359026f/python/vehicle-routing/src/vehicle_routing/json_serialization.py#L9-L14 (note it is not frozen, since the model objects are not used in sets or as keys in dicts).  The frozen requirement for person is because it is used in a set in demo_data.py; you could in theory instead create the sets from the person's ids and then map the ids map back to person.